compile error

```
> node-core-audio@0.5.0 install /Users/tony/src/icecast_client/node_modules/node-core-audio
> node-gyp rebuild

  CXX(target) Release/obj.target/NodeCoreAudio/NodeCoreAudio/AudioEngine.o
In file included from ../NodeCoreAudio/AudioEngine.cpp:9:
In file included from /Users/tony/src/icecast_client/node_modules/node-core-audio/NodeCoreAudio/AudioEngine.h:11:
/Users/tony/.node-gyp/4.2.1/include/node/node_internals.h:6:10: fatal error: 'util-inl.h' file not found
#include "util-inl.h"
         ^
1 error generated.
make: *** [Release/obj.target/NodeCoreAudio/NodeCoreAudio/AudioEngine.o] Error 1
```

I'm using node 4
```
$ node --version
v4.2.1
```

--- a/NodeCoreAudio/AudioEngine.cpp
+++ b/NodeCoreAudio/AudioEngine.cpp
@@ -10,6 +10,7 @@
 #include "portaudio.h"
 #include <v8.h>
 #include <uv.h>
+#include <node.h>
 #include <node_object_wrap.h>
 #include <stdlib.h>
 
@@ -243,7 +244,7 @@ void Audio::AudioEngine::applyOptions( Local<Object> options ) {
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Returns a v8 array filled with input samples */
-Handle<Array> Audio::AudioEngine::getInputBuffer() {
+Local<Array> Audio::AudioEngine::getInputBuffer() {
   Nan::EscapableHandleScope scope;
 
 	if( m_bInterleaved ) {
@@ -274,7 +275,7 @@ Handle<Array> Audio::AudioEngine::getInputBuffer() {
 Handle<Number> Audio::AudioEngine::getSample( int position ) {
   Nan::EscapableHandleScope scope;
 
-	Handle<Number> sample;
+	Local<Number> sample;
 
 	switch( m_uSampleFormat ) {
 	case paFloat32: {
@@ -357,12 +358,12 @@ void Audio::AudioEngine::queueOutputBuffer( Handle<Array> result ) {
 			return;
 		}
 
-		Handle<Array> item;
+		Local<Array> item;
 
 		for( int iChannel=0; iChannel<m_uOutputChannels; ++iChannel ) {
 			for( int iSample=0; iSample<m_uSamplesPerBuffer; ++iSample ) {
 
-				item = Handle<Array>::Cast( result->Get(iChannel) );
+				item = Local<Array>::Cast( result->Get(iChannel) );
 				if( item->IsArray() ) {
 					if( item->Length() > m_uNumCachedOutputSamples[m_uCurrentWriteBuffer] )
 						m_uNumCachedOutputSamples[m_uCurrentWriteBuffer] = item->Length();
@@ -414,9 +415,7 @@ void Audio::AudioEngine::RunAudioLoop(){
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Initialize our node object */
-void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
-
-  Isolate* isolate = target->GetIsolate();
+NAN_MODULE_INIT(Audio::AudioEngine::Init) {
 
 	// Prepare constructor template
 	Local<FunctionTemplate> functionTemplate = Nan::New<FunctionTemplate> (Audio::AudioEngine::New );
@@ -450,7 +449,7 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	//constructor = Persistent<Function>::New( functionTemplate->GetFunction() );
     //Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(EOLFinder::New);
 //    NanAssignPersistent(constructor, functionTemplate->GetFunction());
-    constructor.Reset(isolate, functionTemplate->GetFunction());
+    constructor.Reset(Isolate::GetCurrent(), functionTemplate->GetFunction());
 } // end AudioEngine::Init()
 
 
@@ -519,9 +518,7 @@ void Audio::AudioEngine::write(const Nan::FunctionCallbackInfo<v8::Value>& info)
 	pEngine->queueOutputBuffer( Local<Array>::Cast(info[0]) );
 	uv_mutex_unlock( &pEngine->m_mutex );
 
-	Handle<Boolean> result = Nan::New<Boolean>( false );
-
-	info.GetReturnValue().Set( result );
+	info.GetReturnValue().Set( false );
 } // end AudioEngine::Write()
 
 //////////////////////////////////////////////////////////////////////////////
@@ -533,7 +530,7 @@ void Audio::AudioEngine::isBufferEmpty(const Nan::FunctionCallbackInfo<v8::Value
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
 	uv_mutex_lock( &pEngine->m_mutex );
-	Handle<Boolean> isEmpty = Nan::New<Boolean>(pEngine->m_uNumCachedOutputSamples[pEngine->m_uCurrentWriteBuffer] == 0);
+	Local<Boolean> isEmpty = Nan::New<Boolean>(pEngine->m_uNumCachedOutputSamples[pEngine->m_uCurrentWriteBuffer] == 0);
 	uv_mutex_unlock( &pEngine->m_mutex );
 	info.GetReturnValue().Set( isEmpty );
 } // end AudioEngine::isBufferEmpty()
@@ -550,7 +547,7 @@ void Audio::AudioEngine::read(const Nan::FunctionCallbackInfo<v8::Value>& info) 
 		Pa_ReadStream( pEngine->m_pPaStream, pEngine->m_cachedInputSampleBlock, pEngine->m_uSamplesPerBuffer );
 	}
 
-	Handle<Array> input = pEngine->getInputBuffer();
+	Local<Array> input = pEngine->getInputBuffer();
 
 	info.GetReturnValue().Set( input );
 } // end AudioEngine::Read()

--- a/NodeCoreAudio/AudioEngine.cpp
+++ b/NodeCoreAudio/AudioEngine.cpp
@@ -416,6 +416,9 @@ void Audio::AudioEngine::RunAudioLoop(){
 //////////////////////////////////////////////////////////////////////////////
 /*! Initialize our node object */
 void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
+
+  Isolate* isolate = target->GetIsolate();
+
 	// Prepare constructor template
 	Local<FunctionTemplate> functionTemplate = Nan::New<FunctionTemplate> (Audio::AudioEngine::New );
 	functionTemplate->SetClassName( Nan::New<String>("AudioEngine").ToLocalChecked() );
@@ -448,7 +451,7 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	//constructor = Persistent<Function>::New( functionTemplate->GetFunction() );
     //Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(EOLFinder::New);
 //    NanAssignPersistent(constructor, functionTemplate->GetFunction());
-    constructor.Reset(Nan::GetFunction(functionTemplate));
+    constructor.Reset(isolate, functionTemplate->GetFunction());
 } // end AudioEngine::Init()
 
 

--- a/NodeCoreAudio/AudioEngine.cpp
+++ b/NodeCoreAudio/AudioEngine.cpp
@@ -121,25 +121,23 @@ Audio::AudioEngine::AudioEngine( Local<Object> options ) :
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Gets options */
-//v8::Handle<v8::Value> Audio::AudioEngine::getOptions(const v8::Arguments& args){
 void Audio::AudioEngine::getOptions(const Nan::FunctionCallbackInfo<v8::Value>& info) {
-//NAN_METHOD(Audio::AudioEngine::getOptions){
 	Local<Object> options = Nan::New<Object>();
 
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
-	options->Set( Nan::New<String>("inputChannels"), Nan::New<Number>(pEngine->m_uInputChannels) );
-	options->Set( Nan::New<String>("outputChannels"), Nan::New<Number>(pEngine->m_uOutputChannels) );
+	Nan::Set( options, Nan::New<String>("inputChannels").ToLocalChecked(), Nan::New<Number>(pEngine->m_uInputChannels) );
+	Nan::Set( options, Nan::New<String>("outputChannels").ToLocalChecked(), Nan::New<Number>(pEngine->m_uOutputChannels) );
 
-	options->Set( Nan::New<String>("inputDevice"), Nan::New<Number>(pEngine->m_uInputDevice) );
-	options->Set( Nan::New<String>("outputDevice"), Nan::New<Number>(pEngine->m_uOutputDevice) );
+	Nan::Set( options, Nan::New<String>("inputDevice").ToLocalChecked(), Nan::New<Number>(pEngine->m_uInputDevice) );
+	Nan::Set( options, Nan::New<String>("outputDevice").ToLocalChecked(), Nan::New<Number>(pEngine->m_uOutputDevice) );
 
-	options->Set( Nan::New<String>("sampleRate"), Nan::New<Number>(pEngine->m_uSampleRate) );
-	options->Set( Nan::New<String>("sampleFormat"), Nan::New<Number>(pEngine->m_uSampleFormat) );
-	options->Set( Nan::New<String>("framesPerBuffer"), Nan::New<Number>(pEngine->m_uSamplesPerBuffer) );
-	options->Set( Nan::New<String>("numBuffers"), Nan::New<Number>(pEngine->m_uNumBuffers) );
-	options->Set( Nan::New<String>("interleaved"), Nan::New<Boolean>(pEngine->m_bInterleaved) );
-	options->Set( Nan::New<String>("useMicrophone"), Nan::New<Boolean>(pEngine->m_bReadMicrophone) );
+	Nan::Set( options, Nan::New<String>("sampleRate").ToLocalChecked(), Nan::New<Number>(pEngine->m_uSampleRate) );
+	Nan::Set( options, Nan::New<String>("sampleFormat").ToLocalChecked(), Nan::New<Number>(pEngine->m_uSampleFormat) );
+	Nan::Set( options, Nan::New<String>("framesPerBuffer").ToLocalChecked(), Nan::New<Number>(pEngine->m_uSamplesPerBuffer) );
+	Nan::Set( options, Nan::New<String>("numBuffers").ToLocalChecked(), Nan::New<Number>(pEngine->m_uNumBuffers) );
+	Nan::Set( options, Nan::New<String>("interleaved").ToLocalChecked(), Nan::New<Boolean>(pEngine->m_bInterleaved) );
+	Nan::Set( options, Nan::New<String>("useMicrophone").ToLocalChecked(), Nan::New<Boolean>(pEngine->m_bReadMicrophone) );
 
 	info.GetReturnValue().Set(options);
 } // end GetOptions
@@ -153,11 +151,11 @@ void Audio::AudioEngine::setOptions(const Nan::FunctionCallbackInfo<v8::Value>& 
 	Local<Object> options;
 
 	if( info.Length() > 0 ) {
-		if( !args[0]->IsObject() ) {
+		if( !info[0]->IsObject() ) {
 			return Nan::ThrowTypeError("First argument should be an object.");
 		}
 
-		options = Local<Object>::Cast( args[0] );
+		options = Local<Object>::Cast( info[0] );
 	} else {
         return Nan::ThrowTypeError("First argument does not exist.");
 	}
@@ -173,26 +171,27 @@ void Audio::AudioEngine::setOptions(const Nan::FunctionCallbackInfo<v8::Value>& 
 /*! Sets the given options and restarts the audio stream if necessary */
 void Audio::AudioEngine::applyOptions( Local<Object> options ) {
 	unsigned int oldBufferCount = m_uNumBuffers;
-	if(Nan::HasOwnProperty(options, Nan::New<String>("inputDevice")) )
-		m_uInputDevice = (int)options->Get(Nan::New<String>("inputDevice"))->ToInteger()->Value();
-	if(Nan::HasOwnProperty(options, Nan::New<String>("outputDevice")) )
-		m_uOutputDevice = (int)options->Get(Nan::New<String>("outputDevice"))->ToInteger()->Value();
-	if(Nan::HasOwnProperty(options, Nan::New<String>("inputChannels")) )
-		m_uInputChannels = (int)options->Get(Nan::New<String>("inputChannels"))->ToInteger()->Value();
-	if(Nan::HasOwnProperty(options, Nan::New<String>("outputChannels")) )
-		m_uOutputChannels = (int)options->Get(Nan::New<String>("outputChannels"))->ToInteger()->Value();
-	if(Nan::HasOwnProperty(options, Nan::New<String>("framesPerBuffer")) )
-		m_uSamplesPerBuffer = (int)options->Get(Nan::New<String>("framesPerBuffer"))->ToInteger()->Value();
-	if (Nan::HasOwnProperty(options, Nan::New<String>("numBuffers")) )
-		m_uNumBuffers = (int)options->Get(Nan::New<String>("numBuffers"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("inputDevice").ToLocalChecked()).FromMaybe(false) )
+		m_uInputDevice = Nan::To<int>(Nan::Get(options, Nan::New<String>("inputDevice").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("outputDevice").ToLocalChecked()).FromMaybe(false) )
+		m_uOutputDevice = Nan::To<int>(Nan::Get(options, Nan::New<String>("outputDevice").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("inputChannels").ToLocalChecked()).FromMaybe(false) )
+		m_uInputChannels = Nan::To<int>(Nan::Get(options, Nan::New<String>("inputChannels").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("outputChannels").ToLocalChecked()).FromMaybe(false) )
+	  m_uOutputChannels =
+	    Nan::To<int>(Nan::Get(options, Nan::New<String>("outputChannels").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("framesPerBuffer").ToLocalChecked()).FromMaybe(false) )
+		m_uSamplesPerBuffer = Nan::To<int>(Nan::Get(options, Nan::New<String>("framesPerBuffer").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if (Nan::HasOwnProperty(options, Nan::New<String>("numBuffers").ToLocalChecked()).FromMaybe(false) )
+		m_uNumBuffers = Nan::To<int>(Nan::Get(options, Nan::New<String>("numBuffers").ToLocalChecked()).ToLocalChecked()).FromJust();
 
-	if(Nan::HasOwnProperty(options, Nan::New<String>("interleaved")) )
-		m_bInterleaved = options->Get(Nan::New<String>("interleaved"))->ToBoolean()->Value();
-	if (Nan::HasOwnProperty(options, Nan::New<String>("useMicrophone")) )
-		m_bReadMicrophone = options->Get(Nan::New<String>("useMicrophone"))->ToBoolean()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("interleaved").ToLocalChecked()).FromMaybe(false) )
+		m_bInterleaved = Nan::To<bool>(Nan::Get(options, Nan::New<String>("interleaved").ToLocalChecked()).ToLocalChecked()).FromJust();
+	if (Nan::HasOwnProperty(options, Nan::New<String>("useMicrophone").ToLocalChecked()).FromMaybe(false) )
+		m_bReadMicrophone = Nan::To<bool>(Nan::Get(options, Nan::New<String>("useMicrophone").ToLocalChecked()).ToLocalChecked()).FromJust();;
 
-	if(Nan::HasOwnProperty(options, Nan::New<String>("sampleFormat")) ) {
-		switch( options->Get(Nan::New<String>("sampleFormat"))->ToInteger()->Value() ){
+	if(Nan::HasOwnProperty(options, Nan::New<String>("sampleFormat").ToLocalChecked()).FromMaybe(false) ) {
+		switch(Nan::To<int>(Nan::Get(options, Nan::New<String>("sampleFormat").ToLocalChecked()).ToLocalChecked()).FromJust()){
 			case 0x01: m_uSampleFormat = paFloat32; m_uSampleSize = 4; break;
 			case 0x02: m_uSampleFormat = paInt32; m_uSampleSize = 4; break;
 			case 0x04: m_uSampleFormat = paInt24; m_uSampleSize = 3; break;
@@ -246,10 +245,7 @@ void Audio::AudioEngine::applyOptions( Local<Object> options ) {
 //////////////////////////////////////////////////////////////////////////////
 /*! Returns a v8 array filled with input samples */
 Handle<Array> Audio::AudioEngine::getInputBuffer() {
-//NAN_METHOD(Audio::AudioEngine::getInputBuffer){
-    NanEscapableScope();
-    //NanScope();
-	//HandleScope scope;
+  Nan::EscapableHandleScope scope;
 
 	if( m_bInterleaved ) {
 		m_hInputBuffer = Nan::New<Array>( m_uInputChannels * m_uSamplesPerBuffer );
@@ -270,16 +266,14 @@ Handle<Array> Audio::AudioEngine::getInputBuffer() {
 		}
 	}
 
-	return NanEscapeScope( m_hInputBuffer );
+	return scope.Escape( m_hInputBuffer );
 } // end AudioEngine::getInputBuffer()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Returns a sound card sample converted to a v8 Number */
 Handle<Number> Audio::AudioEngine::getSample( int position ) {
-//NAN_METHOD(Audio::AudioEngine::getSample){
-    NanEscapableScope();
-	//HandleScope scope;
+  Nan::EscapableHandleScope scope;
 
 	Handle<Number> sample;
 
@@ -310,7 +304,7 @@ Handle<Number> Audio::AudioEngine::getSample( int position ) {
 		break;
 	}
 
-    return NanEscapeScope(sample);
+    return scope.Escape(sample);
 } // end AudioEngine::getSample()
 
 
@@ -424,7 +418,7 @@ void Audio::AudioEngine::RunAudioLoop(){
 void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	// Prepare constructor template
 	Local<FunctionTemplate> functionTemplate = Nan::New<FunctionTemplate> (Audio::AudioEngine::New );
-	functionTemplate->SetClassName( Nan::New<String>("AudioEngine") );
+	functionTemplate->SetClassName( Nan::New<String>("AudioEngine").ToLocalChecked() );
 	functionTemplate->InstanceTemplate()->SetInternalFieldCount( 1 );
 
 
@@ -435,9 +429,9 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("isActive"), Nan::New<FunctionTemplate>(Audio::AudioEngine::isActive)->GetFunction() );
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("getDeviceName"), Nan::New<FunctionTemplate>(Audio::AudioEngine::getDeviceName)->GetFunction() );
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("getNumDevices"), Nan::New<FunctionTemplate>(Audio::AudioEngine::getNumDevices)->GetFunction() );
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "isActive", Audio::AudioEngine::isActive);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getDeviceName", Audio::AudioEngine::getDeviceName);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getNumDevices", Audio::AudioEngine::getNumDevices);
+    Nan::SetPrototypeMethod(functionTemplate, "isActive", Audio::AudioEngine::isActive);
+    Nan::SetPrototypeMethod(functionTemplate, "getDeviceName", Audio::AudioEngine::getDeviceName);
+    Nan::SetPrototypeMethod(functionTemplate, "getNumDevices", Audio::AudioEngine::getNumDevices);
 
 	// Set
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("setOptions"), Nan::New<FunctionTemplate>(Audio::AudioEngine::setOptions)->GetFunction() );
@@ -445,114 +439,110 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("write"), Nan::New<FunctionTemplate>(Audio::AudioEngine::write)->GetFunction() );
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("read"), Nan::New<FunctionTemplate>(Audio::AudioEngine::read)->GetFunction() );
 	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("isBufferEmpty"), Nan::New<FunctionTemplate>(Audio::AudioEngine::isBufferEmpty)->GetFunction() );
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "setOptions", Audio::AudioEngine::setOptions);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getOptions", Audio::AudioEngine::getOptions);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "write", Audio::AudioEngine::write);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "read", Audio::AudioEngine::read);
-    NODE_SET_PROTOTYPE_METHOD(functionTemplate, "isBufferEmpty", Audio::AudioEngine::isBufferEmpty);
+    Nan::SetPrototypeMethod(functionTemplate, "setOptions", Audio::AudioEngine::setOptions);
+    Nan::SetPrototypeMethod(functionTemplate, "getOptions", Audio::AudioEngine::getOptions);
+    Nan::SetPrototypeMethod(functionTemplate, "write", Audio::AudioEngine::write);
+    Nan::SetPrototypeMethod(functionTemplate, "read", Audio::AudioEngine::read);
+    Nan::SetPrototypeMethod(functionTemplate, "isBufferEmpty", Audio::AudioEngine::isBufferEmpty);
 
 	//constructor = Persistent<Function>::New( functionTemplate->GetFunction() );
     //Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(EOLFinder::New);
-    NanAssignPersistent(constructor, functionTemplate->GetFunction());
+//    NanAssignPersistent(constructor, functionTemplate->GetFunction());
+    constructor.Reset(Nan::GetFunction(functionTemplate));
 } // end AudioEngine::Init()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Create a new instance of the audio engine */
-//v8::Handle<v8::Value> Audio::AudioEngine::NewInstance(const v8::Arguments& args) {
-
-NAN_METHOD(Audio::AudioEngine::NewInstance){
+//v8::Handle<v8::Value> Audio::AudioEngine::NewInstance(const v8::Arguments& info) {
+void Audio::AudioEngine::NewInstance(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	unsigned argc = args.Length();
+	unsigned argc = info.Length();
 
 	if( argc > 2 )
 		argc = 2;
 
 	Handle<Value>* argv = new Handle<Value>[argc];
 
-	argv[0] = args[0];
+	argv[0] = info[0];
 	if( argc > 1 )
-		argv[1] = args[1];
+		argv[1] = info[1];
 
 	//Local<Object> instance = constructor->NewInstance( argc, argv );
 	Local<Object> instance = Nan::New(constructor)->NewInstance(argc, argv);
 	//Local<Object> instance = constructor->NewInstance(argc, argv);
 
-	Nan::ReturnValue( instance );
+	info.GetReturnValue().Set( instance );
 } // end AudioEngine::NewInstance()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Create a v8 object */
-//v8::Handle<v8::Value> Audio::AudioEngine::New( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::New){
+void Audio::AudioEngine::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
 	Local<Object> options;
 
-	if( args.Length() > 0 ) {
-		if( !args[0]->IsObject() )
+	if( info.Length() > 0 ) {
+		if( !info[0]->IsObject() )
             return Nan::ThrowTypeError("First argument must be an object.");
 		else
-			options = Local<Object>::Cast( args[0] );
+			options = Local<Object>::Cast( info[0] );
 	} else {
 		options = Nan::New<Object>();
 	}
 
 	AudioEngine* pEngine = new AudioEngine( options );
-	pEngine->Wrap( args.This() );
+	pEngine->Wrap( info.This() );
 
-	Nan::ReturnValue( args.This() );
+	info.GetReturnValue().Set( info.This() );
 } // end AudioEngine::New()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Write samples to the current audio device */
-//v8::Handle<v8::Value> Audio::AudioEngine::write( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::write){
+void Audio::AudioEngine::write(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
-	if (args.Length() > 1 || !args[0]->IsArray()){
+	if (info.Length() > 1 || !info[0]->IsArray()){
         return Nan::ThrowTypeError("First argument should be an array.");
 	}
 
 	uv_mutex_lock( &pEngine->m_mutex );
-	pEngine->queueOutputBuffer( Local<Array>::Cast(args[0]) );
+	pEngine->queueOutputBuffer( Local<Array>::Cast(info[0]) );
 	uv_mutex_unlock( &pEngine->m_mutex );
 
 	Handle<Boolean> result = Nan::New<Boolean>( false );
 
-	Nan::ReturnValue( result );
+	info.GetReturnValue().Set( result );
 } // end AudioEngine::Write()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Checks if the current audio buffer has been fed to Port Audio */
-//v8::Handle<v8::Value> Audio::AudioEngine::isBufferEmpty( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::isBufferEmpty){
+void Audio::AudioEngine::isBufferEmpty(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
 	uv_mutex_lock( &pEngine->m_mutex );
 	Handle<Boolean> isEmpty = Nan::New<Boolean>(pEngine->m_uNumCachedOutputSamples[pEngine->m_uCurrentWriteBuffer] == 0);
 	uv_mutex_unlock( &pEngine->m_mutex );
-	Nan::ReturnValue( isEmpty );
+	info.GetReturnValue().Set( isEmpty );
 } // end AudioEngine::isBufferEmpty()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Read samples from the current audio device */
-//v8::Handle<v8::Value> Audio::AudioEngine::read( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::read){
+void Audio::AudioEngine::read(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
 	if (pEngine->m_bReadMicrophone) {
 		Pa_ReadStream( pEngine->m_pPaStream, pEngine->m_cachedInputSampleBlock, pEngine->m_uSamplesPerBuffer );
@@ -560,55 +550,52 @@ NAN_METHOD(Audio::AudioEngine::read){
 
 	Handle<Array> input = pEngine->getInputBuffer();
 
-	Nan::ReturnValue( input );
+	info.GetReturnValue().Set( input );
 } // end AudioEngine::Read()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Returns whether the PortAudio stream is active */
-//v8::Handle<v8::Value> Audio::AudioEngine::isActive( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::isActive){
+void Audio::AudioEngine::isActive(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
 	if( Pa_IsStreamActive(pEngine->m_pPaStream) )
-		Nan::ReturnValue( Nan::New<Boolean>(true) );
+		info.GetReturnValue().Set( Nan::New<Boolean>(true) );
 	else
-		Nan::ReturnValue( Nan::New<Boolean>(false) );
+		info.GetReturnValue().Set( Nan::New<Boolean>(false) );
 } // end AudioEngine::IsActive()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Get the name of an audio device with a given ID number */
-//v8::Handle<v8::Value> Audio::AudioEngine::getDeviceName( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::getDeviceName){
+void Audio::AudioEngine::getDeviceName(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
-	if( !args[0]->IsNumber() ) {
+	if( !info[0]->IsNumber() ) {
 		return  Nan::ThrowTypeError("getDeviceName() requires a device index");
 	}
 
-	Local<Number> deviceIndex = Local<Number>::Cast( args[0] );
+	Local<Number> deviceIndex = Local<Number>::Cast( info[0] );
 
 	const PaDeviceInfo* pDeviceInfo = Pa_GetDeviceInfo( (PaDeviceIndex)deviceIndex->NumberValue() );
 
-	Nan::ReturnValue( Nan::New<String>(pDeviceInfo->name) );
+	info.GetReturnValue().Set( Nan::New<String>(pDeviceInfo->name).ToLocalChecked() );
 } // end AudioEngine::GetDeviceName()
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Get the number of available devices */
-//v8::Handle<v8::Value> Audio::AudioEngine::getNumDevices( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::getNumDevices){
+void Audio::AudioEngine::getNumDevices(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     Nan::HandleScope scope;
 	//HandleScope scope;
 
 	int deviceCount = Pa_GetDeviceCount();
 
-	Nan::ReturnValue( Nan::New<Number>(deviceCount) );
+	info.GetReturnValue().Set( Nan::New<Number>(deviceCount) );
 } // end AudioEngine::GetNumDevices()
 
 

--- a/NodeCoreAudio/AudioEngine.cpp
+++ b/NodeCoreAudio/AudioEngine.cpp
@@ -62,25 +62,25 @@ Audio::AudioEngine::AudioEngine( Local<Object> options ) :
 	m_uCurrentReadBuffer = 0;
 
 	// Create V8 objects to hold our buffers
-	m_hInputBuffer = NanNew<Array>( m_uInputChannels );
+	m_hInputBuffer = Nan::New<Array>( m_uInputChannels );
 	for( int iChannel=0; iChannel<m_uInputChannels; iChannel++ )
-		m_hInputBuffer->Set( iChannel, NanNew<Array>(m_uSamplesPerBuffer) );
+		m_hInputBuffer->Set( iChannel, Nan::New<Array>(m_uSamplesPerBuffer) );
 
 	// Initialize our audio core
 	PaError initErr = Pa_Initialize();
 
 	m_uInputDevice = Pa_GetDefaultInputDevice();
 	if( m_uInputDevice == paNoDevice ) {
-		NanThrowTypeError("Error: No default input device");
+		Nan::ThrowTypeError("Error: No default input device");
 	};
 
 	m_uOutputDevice = Pa_GetDefaultOutputDevice();
 	if( m_uOutputDevice == paNoDevice ) {
-		NanThrowTypeError("Error: No default output device");
+		Nan::ThrowTypeError("Error: No default output device");
 	}
 
 	if( initErr != paNoError )
-		NanThrowTypeError("Failed to initialize audio engine");
+		Nan::ThrowTypeError("Failed to initialize audio engine");
 
 	applyOptions( options );
 
@@ -105,13 +105,13 @@ Audio::AudioEngine::AudioEngine( Local<Object> options ) :
 									NULL );
 
 	if( openStreamErr != paNoError )
-		NanThrowTypeError("Failed to open audio stream");
+		Nan::ThrowTypeError("Failed to open audio stream");
 
 	// Start the audio stream
 	PaError startStreamErr = Pa_StartStream( m_pPaStream );
 
 	if( startStreamErr != paNoError )
-		NanThrowTypeError("Failed to start audio stream");
+		Nan::ThrowTypeError("Failed to start audio stream");
 
 	uv_mutex_init( &m_mutex );
 	uv_thread_create( &ptStreamThread, do_work, (void*)this );
@@ -122,54 +122,50 @@ Audio::AudioEngine::AudioEngine( Local<Object> options ) :
 //////////////////////////////////////////////////////////////////////////////
 /*! Gets options */
 //v8::Handle<v8::Value> Audio::AudioEngine::getOptions(const v8::Arguments& args){
-NAN_METHOD(Audio::AudioEngine::getOptions){
-    NanScope();
-	//HandleScope scope;
-    NanLocker();
+void Audio::AudioEngine::getOptions(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+//NAN_METHOD(Audio::AudioEngine::getOptions){
+	Local<Object> options = Nan::New<Object>();
 
-	Local<Object> options = NanNew<Object>();
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	options->Set( Nan::New<String>("inputChannels"), Nan::New<Number>(pEngine->m_uInputChannels) );
+	options->Set( Nan::New<String>("outputChannels"), Nan::New<Number>(pEngine->m_uOutputChannels) );
 
-	options->Set( NanNew<String>("inputChannels"), NanNew<Number>(pEngine->m_uInputChannels) );
-	options->Set( NanNew<String>("outputChannels"), NanNew<Number>(pEngine->m_uOutputChannels) );
+	options->Set( Nan::New<String>("inputDevice"), Nan::New<Number>(pEngine->m_uInputDevice) );
+	options->Set( Nan::New<String>("outputDevice"), Nan::New<Number>(pEngine->m_uOutputDevice) );
 
-	options->Set( NanNew<String>("inputDevice"), NanNew<Number>(pEngine->m_uInputDevice) );
-	options->Set( NanNew<String>("outputDevice"), NanNew<Number>(pEngine->m_uOutputDevice) );
+	options->Set( Nan::New<String>("sampleRate"), Nan::New<Number>(pEngine->m_uSampleRate) );
+	options->Set( Nan::New<String>("sampleFormat"), Nan::New<Number>(pEngine->m_uSampleFormat) );
+	options->Set( Nan::New<String>("framesPerBuffer"), Nan::New<Number>(pEngine->m_uSamplesPerBuffer) );
+	options->Set( Nan::New<String>("numBuffers"), Nan::New<Number>(pEngine->m_uNumBuffers) );
+	options->Set( Nan::New<String>("interleaved"), Nan::New<Boolean>(pEngine->m_bInterleaved) );
+	options->Set( Nan::New<String>("useMicrophone"), Nan::New<Boolean>(pEngine->m_bReadMicrophone) );
 
-	options->Set( NanNew<String>("sampleRate"), NanNew<Number>(pEngine->m_uSampleRate) );
-	options->Set( NanNew<String>("sampleFormat"), NanNew<Number>(pEngine->m_uSampleFormat) );
-	options->Set( NanNew<String>("framesPerBuffer"), NanNew<Number>(pEngine->m_uSamplesPerBuffer) );
-	options->Set( NanNew<String>("numBuffers"), NanNew<Number>(pEngine->m_uNumBuffers) );
-	options->Set( NanNew<String>("interleaved"), NanNew<Boolean>(pEngine->m_bInterleaved) );
-	options->Set( NanNew<String>("useMicrophone"), NanNew<Boolean>(pEngine->m_bReadMicrophone) );
-
-	NanReturnValue(options);
+	info.GetReturnValue().Set(options);
 } // end GetOptions
 
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Set options, restarts audio stream */
-//v8::Handle<v8::Value> Audio::AudioEngine::setOptions( const v8::Arguments& args ) {
-NAN_METHOD(Audio::AudioEngine::setOptions){
-    NanScope();
+void Audio::AudioEngine::setOptions(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    Nan::HandleScope scope;
 	//HandleScope scope;
 	Local<Object> options;
 
-	if( args.Length() > 0 ) {
+	if( info.Length() > 0 ) {
 		if( !args[0]->IsObject() ) {
-			return NanThrowTypeError("First argument should be an object.");
+			return Nan::ThrowTypeError("First argument should be an object.");
 		}
 
 		options = Local<Object>::Cast( args[0] );
 	} else {
-        return NanThrowTypeError("First argument does not exist.");
+        return Nan::ThrowTypeError("First argument does not exist.");
 	}
 
-	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
+	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( info.This() );
 	pEngine->applyOptions( options );
 
-    NanReturnUndefined();
+    	info.GetReturnValue().SetUndefined();
 } // end AudioEngine::SetOptions()x
 
 
@@ -177,26 +173,26 @@ NAN_METHOD(Audio::AudioEngine::setOptions){
 /*! Sets the given options and restarts the audio stream if necessary */
 void Audio::AudioEngine::applyOptions( Local<Object> options ) {
 	unsigned int oldBufferCount = m_uNumBuffers;
-	if( options->HasOwnProperty(NanNew<String>("inputDevice")) )
-		m_uInputDevice = (int)options->Get(NanNew<String>("inputDevice"))->ToInteger()->Value();
-	if( options->HasOwnProperty(NanNew<String>("outputDevice")) )
-		m_uOutputDevice = (int)options->Get(NanNew<String>("outputDevice"))->ToInteger()->Value();
-	if( options->HasOwnProperty(NanNew<String>("inputChannels")) )
-		m_uInputChannels = (int)options->Get(NanNew<String>("inputChannels"))->ToInteger()->Value();
-	if( options->HasOwnProperty(NanNew<String>("outputChannels")) )
-		m_uOutputChannels = (int)options->Get(NanNew<String>("outputChannels"))->ToInteger()->Value();
-	if( options->HasOwnProperty(NanNew<String>("framesPerBuffer")) )
-		m_uSamplesPerBuffer = (int)options->Get(NanNew<String>("framesPerBuffer"))->ToInteger()->Value();
-	if ( options->HasOwnProperty(NanNew<String>("numBuffers")) )
-		m_uNumBuffers = (int)options->Get(NanNew<String>("numBuffers"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("inputDevice")) )
+		m_uInputDevice = (int)options->Get(Nan::New<String>("inputDevice"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("outputDevice")) )
+		m_uOutputDevice = (int)options->Get(Nan::New<String>("outputDevice"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("inputChannels")) )
+		m_uInputChannels = (int)options->Get(Nan::New<String>("inputChannels"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("outputChannels")) )
+		m_uOutputChannels = (int)options->Get(Nan::New<String>("outputChannels"))->ToInteger()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("framesPerBuffer")) )
+		m_uSamplesPerBuffer = (int)options->Get(Nan::New<String>("framesPerBuffer"))->ToInteger()->Value();
+	if (Nan::HasOwnProperty(options, Nan::New<String>("numBuffers")) )
+		m_uNumBuffers = (int)options->Get(Nan::New<String>("numBuffers"))->ToInteger()->Value();
 
-	if( options->HasOwnProperty(NanNew<String>("interleaved")) )
-		m_bInterleaved = options->Get(NanNew<String>("interleaved"))->ToBoolean()->Value();
-	if ( options->HasOwnProperty(NanNew<String>("useMicrophone")) )
-		m_bReadMicrophone = options->Get(NanNew<String>("useMicrophone"))->ToBoolean()->Value();
+	if(Nan::HasOwnProperty(options, Nan::New<String>("interleaved")) )
+		m_bInterleaved = options->Get(Nan::New<String>("interleaved"))->ToBoolean()->Value();
+	if (Nan::HasOwnProperty(options, Nan::New<String>("useMicrophone")) )
+		m_bReadMicrophone = options->Get(Nan::New<String>("useMicrophone"))->ToBoolean()->Value();
 
-	if( options->HasOwnProperty(NanNew<String>("sampleFormat")) ) {
-		switch( options->Get(NanNew<String>("sampleFormat"))->ToInteger()->Value() ){
+	if(Nan::HasOwnProperty(options, Nan::New<String>("sampleFormat")) ) {
+		switch( options->Get(Nan::New<String>("sampleFormat"))->ToInteger()->Value() ){
 			case 0x01: m_uSampleFormat = paFloat32; m_uSampleSize = 4; break;
 			case 0x02: m_uSampleFormat = paInt32; m_uSampleSize = 4; break;
 			case 0x04: m_uSampleFormat = paInt24; m_uSampleSize = 3; break;
@@ -253,19 +249,18 @@ Handle<Array> Audio::AudioEngine::getInputBuffer() {
 //NAN_METHOD(Audio::AudioEngine::getInputBuffer){
     NanEscapableScope();
     //NanScope();
-    NanLocker();
 	//HandleScope scope;
 
 	if( m_bInterleaved ) {
-		m_hInputBuffer = NanNew<Array>( m_uInputChannels * m_uSamplesPerBuffer );
+		m_hInputBuffer = Nan::New<Array>( m_uInputChannels * m_uSamplesPerBuffer );
 
 		for( int iSample=0; iSample<m_uSamplesPerBuffer*m_uInputChannels; ++iSample ) {
 			m_hInputBuffer->Set( iSample, getSample(iSample) );
 		}
 	} else {
-		m_hInputBuffer = NanNew<Array>( m_uInputChannels );
+		m_hInputBuffer = Nan::New<Array>( m_uInputChannels );
 		for( int iChannel=0; iChannel<m_uInputChannels; iChannel++ ) {
-			auto tempBuffer = Local<Array>( NanNew<Array>(m_uSamplesPerBuffer) );
+			auto tempBuffer = Local<Array>( Nan::New<Array>(m_uSamplesPerBuffer) );
 
 			for( int iSample=0; iSample<m_uSamplesPerBuffer; iSample++ ) {
 				tempBuffer->Set( iSample, getSample(iSample) );
@@ -284,7 +279,6 @@ Handle<Array> Audio::AudioEngine::getInputBuffer() {
 Handle<Number> Audio::AudioEngine::getSample( int position ) {
 //NAN_METHOD(Audio::AudioEngine::getSample){
     NanEscapableScope();
-    NanLocker();
 	//HandleScope scope;
 
 	Handle<Number> sample;
@@ -293,26 +287,26 @@ Handle<Number> Audio::AudioEngine::getSample( int position ) {
 	case paFloat32: {
 		float fValue = ((float*)m_cachedInputSampleBlock)[position];
 
-		sample = NanNew<Number>( fValue );
+		sample = Nan::New<Number>( fValue );
 		break;
 					}
 	case paInt32:
-		sample = NanNew<Integer>( ((int*)m_cachedInputSampleBlock)[position] );
+		sample = Nan::New<Integer>( ((int*)m_cachedInputSampleBlock)[position] );
 		break;
 
 	case paInt24:
-		sample = NanNew<Integer>(
+		sample = Nan::New<Integer>(
 				(m_cachedInputSampleBlock[3*position + 0] << 16)
 			  + (m_cachedInputSampleBlock[3*position + 1] << 8)
 			  + (m_cachedInputSampleBlock[3*position + 2]) );
 		break;
 
 	case paInt16:
-		sample = NanNew<Integer>( ((int16_t*)m_cachedInputSampleBlock)[position] );
+		sample = Nan::New<Integer>( ((int16_t*)m_cachedInputSampleBlock)[position] );
 		break;
 
 	default:
-		sample = NanNew<Integer>( m_cachedInputSampleBlock[position]*-1 );
+		sample = Nan::New<Integer>( m_cachedInputSampleBlock[position]*-1 );
 		break;
 	}
 
@@ -366,7 +360,7 @@ void Audio::AudioEngine::queueOutputBuffer( Handle<Array> result ) {
 	} else {
 		// Validate the structure of the output buffer array
 		if( !result->Get(0)->IsArray() ) {
-			NanThrowTypeError("Output buffer not properly setup, 0th channel is not an array");
+			Nan::ThrowTypeError("Output buffer not properly setup, 0th channel is not an array");
 			return;
 		}
 
@@ -429,28 +423,28 @@ void Audio::AudioEngine::RunAudioLoop(){
 /*! Initialize our node object */
 void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 	// Prepare constructor template
-	Local<FunctionTemplate> functionTemplate = NanNew<FunctionTemplate> (Audio::AudioEngine::New );
-	functionTemplate->SetClassName( NanNew<String>("AudioEngine") );
+	Local<FunctionTemplate> functionTemplate = Nan::New<FunctionTemplate> (Audio::AudioEngine::New );
+	functionTemplate->SetClassName( Nan::New<String>("AudioEngine") );
 	functionTemplate->InstanceTemplate()->SetInternalFieldCount( 1 );
 
 
-    //Local<FunctionTemplate> constructorHandle = NanNew(constructor);
-    //target->Set(NanNew<String>("AudioEngine"), functionTemplate->GetFunction());
+    //Local<FunctionTemplate> constructorHandle = Nan::New(constructor);
+    //target->Set(Nan::New<String>("AudioEngine"), functionTemplate->GetFunction());
 	
     // Get
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("isActive"), NanNew<FunctionTemplate>(Audio::AudioEngine::isActive)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("getDeviceName"), NanNew<FunctionTemplate>(Audio::AudioEngine::getDeviceName)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("getNumDevices"), NanNew<FunctionTemplate>(Audio::AudioEngine::getNumDevices)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("isActive"), Nan::New<FunctionTemplate>(Audio::AudioEngine::isActive)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("getDeviceName"), Nan::New<FunctionTemplate>(Audio::AudioEngine::getDeviceName)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("getNumDevices"), Nan::New<FunctionTemplate>(Audio::AudioEngine::getNumDevices)->GetFunction() );
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "isActive", Audio::AudioEngine::isActive);
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getDeviceName", Audio::AudioEngine::getDeviceName);
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getNumDevices", Audio::AudioEngine::getNumDevices);
 
 	// Set
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("setOptions"), NanNew<FunctionTemplate>(Audio::AudioEngine::setOptions)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("getOptions"), NanNew<FunctionTemplate>(Audio::AudioEngine::getOptions)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("write"), NanNew<FunctionTemplate>(Audio::AudioEngine::write)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("read"), NanNew<FunctionTemplate>(Audio::AudioEngine::read)->GetFunction() );
-	//functionTemplate->PrototypeTemplate()->Set( NanNew<String>("isBufferEmpty"), NanNew<FunctionTemplate>(Audio::AudioEngine::isBufferEmpty)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("setOptions"), Nan::New<FunctionTemplate>(Audio::AudioEngine::setOptions)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("getOptions"), Nan::New<FunctionTemplate>(Audio::AudioEngine::getOptions)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("write"), Nan::New<FunctionTemplate>(Audio::AudioEngine::write)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("read"), Nan::New<FunctionTemplate>(Audio::AudioEngine::read)->GetFunction() );
+	//functionTemplate->PrototypeTemplate()->Set( Nan::New<String>("isBufferEmpty"), Nan::New<FunctionTemplate>(Audio::AudioEngine::isBufferEmpty)->GetFunction() );
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "setOptions", Audio::AudioEngine::setOptions);
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "getOptions", Audio::AudioEngine::getOptions);
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "write", Audio::AudioEngine::write);
@@ -458,7 +452,7 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
     NODE_SET_PROTOTYPE_METHOD(functionTemplate, "isBufferEmpty", Audio::AudioEngine::isBufferEmpty);
 
 	//constructor = Persistent<Function>::New( functionTemplate->GetFunction() );
-    //Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(EOLFinder::New);
+    //Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(EOLFinder::New);
     NanAssignPersistent(constructor, functionTemplate->GetFunction());
 } // end AudioEngine::Init()
 
@@ -468,7 +462,7 @@ void Audio::AudioEngine::Init( v8::Handle<v8::Object> target ) {
 //v8::Handle<v8::Value> Audio::AudioEngine::NewInstance(const v8::Arguments& args) {
 
 NAN_METHOD(Audio::AudioEngine::NewInstance){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	unsigned argc = args.Length();
@@ -483,34 +477,34 @@ NAN_METHOD(Audio::AudioEngine::NewInstance){
 		argv[1] = args[1];
 
 	//Local<Object> instance = constructor->NewInstance( argc, argv );
-	Local<Object> instance = NanNew(constructor)->NewInstance(argc, argv);
+	Local<Object> instance = Nan::New(constructor)->NewInstance(argc, argv);
 	//Local<Object> instance = constructor->NewInstance(argc, argv);
 
-	NanReturnValue( instance );
+	Nan::ReturnValue( instance );
 } // end AudioEngine::NewInstance()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Create a v8 object */
 //v8::Handle<v8::Value> Audio::AudioEngine::New( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::New){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	Local<Object> options;
 
 	if( args.Length() > 0 ) {
 		if( !args[0]->IsObject() )
-            return NanThrowTypeError("First argument must be an object.");
+            return Nan::ThrowTypeError("First argument must be an object.");
 		else
 			options = Local<Object>::Cast( args[0] );
 	} else {
-		options = NanNew<Object>();
+		options = Nan::New<Object>();
 	}
 
 	AudioEngine* pEngine = new AudioEngine( options );
 	pEngine->Wrap( args.This() );
 
-	NanReturnValue( args.This() );
+	Nan::ReturnValue( args.This() );
 } // end AudioEngine::New()
 
 
@@ -518,44 +512,44 @@ NAN_METHOD(Audio::AudioEngine::New){
 /*! Write samples to the current audio device */
 //v8::Handle<v8::Value> Audio::AudioEngine::write( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::write){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
 
 	if (args.Length() > 1 || !args[0]->IsArray()){
-        return NanThrowTypeError("First argument should be an array.");
+        return Nan::ThrowTypeError("First argument should be an array.");
 	}
 
 	uv_mutex_lock( &pEngine->m_mutex );
 	pEngine->queueOutputBuffer( Local<Array>::Cast(args[0]) );
 	uv_mutex_unlock( &pEngine->m_mutex );
 
-	Handle<Boolean> result = NanNew<Boolean>( false );
+	Handle<Boolean> result = Nan::New<Boolean>( false );
 
-	NanReturnValue( result );
+	Nan::ReturnValue( result );
 } // end AudioEngine::Write()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Checks if the current audio buffer has been fed to Port Audio */
 //v8::Handle<v8::Value> Audio::AudioEngine::isBufferEmpty( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::isBufferEmpty){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
 
 	uv_mutex_lock( &pEngine->m_mutex );
-	Handle<Boolean> isEmpty = NanNew<Boolean>(pEngine->m_uNumCachedOutputSamples[pEngine->m_uCurrentWriteBuffer] == 0);
+	Handle<Boolean> isEmpty = Nan::New<Boolean>(pEngine->m_uNumCachedOutputSamples[pEngine->m_uCurrentWriteBuffer] == 0);
 	uv_mutex_unlock( &pEngine->m_mutex );
-	NanReturnValue( isEmpty );
+	Nan::ReturnValue( isEmpty );
 } // end AudioEngine::isBufferEmpty()
 
 //////////////////////////////////////////////////////////////////////////////
 /*! Read samples from the current audio device */
 //v8::Handle<v8::Value> Audio::AudioEngine::read( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::read){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
@@ -566,7 +560,7 @@ NAN_METHOD(Audio::AudioEngine::read){
 
 	Handle<Array> input = pEngine->getInputBuffer();
 
-	NanReturnValue( input );
+	Nan::ReturnValue( input );
 } // end AudioEngine::Read()
 
 
@@ -574,15 +568,15 @@ NAN_METHOD(Audio::AudioEngine::read){
 /*! Returns whether the PortAudio stream is active */
 //v8::Handle<v8::Value> Audio::AudioEngine::isActive( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::isActive){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	AudioEngine* pEngine = AudioEngine::Unwrap<AudioEngine>( args.This() );
 
 	if( Pa_IsStreamActive(pEngine->m_pPaStream) )
-		NanReturnValue( NanNew<Boolean>(true) );
+		Nan::ReturnValue( Nan::New<Boolean>(true) );
 	else
-		NanReturnValue( NanNew<Boolean>(false) );
+		Nan::ReturnValue( Nan::New<Boolean>(false) );
 } // end AudioEngine::IsActive()
 
 
@@ -590,18 +584,18 @@ NAN_METHOD(Audio::AudioEngine::isActive){
 /*! Get the name of an audio device with a given ID number */
 //v8::Handle<v8::Value> Audio::AudioEngine::getDeviceName( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::getDeviceName){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	if( !args[0]->IsNumber() ) {
-		return  NanThrowTypeError("getDeviceName() requires a device index");
+		return  Nan::ThrowTypeError("getDeviceName() requires a device index");
 	}
 
 	Local<Number> deviceIndex = Local<Number>::Cast( args[0] );
 
 	const PaDeviceInfo* pDeviceInfo = Pa_GetDeviceInfo( (PaDeviceIndex)deviceIndex->NumberValue() );
 
-	NanReturnValue( NanNew<String>(pDeviceInfo->name) );
+	Nan::ReturnValue( Nan::New<String>(pDeviceInfo->name) );
 } // end AudioEngine::GetDeviceName()
 
 
@@ -609,12 +603,12 @@ NAN_METHOD(Audio::AudioEngine::getDeviceName){
 /*! Get the number of available devices */
 //v8::Handle<v8::Value> Audio::AudioEngine::getNumDevices( const v8::Arguments& args ) {
 NAN_METHOD(Audio::AudioEngine::getNumDevices){
-    NanScope();
+    Nan::HandleScope scope;
 	//HandleScope scope;
 
 	int deviceCount = Pa_GetDeviceCount();
 
-	NanReturnValue( NanNew<Number>(deviceCount) );
+	Nan::ReturnValue( Nan::New<Number>(deviceCount) );
 } // end AudioEngine::GetNumDevices()
 
 
@@ -627,15 +621,15 @@ void Audio::AudioEngine::restartStream() {
 	error = Pa_StopStream( m_pPaStream );
 
 	if( error != paNoError )
-		NanThrowTypeError("Failed to stop audio stream");
+		Nan::ThrowTypeError("Failed to stop audio stream");
 
 	// Close the audio stream
 	error = Pa_CloseStream( m_pPaStream );
 
 	if( error != paNoError )
-		NanThrowTypeError("Failed to close audio stream");
+		Nan::ThrowTypeError("Failed to close audio stream");
 
-	// Open an audio stream.
+	// Open an audio stream.``
 	error = Pa_OpenStream(  &m_pPaStream,
 		&m_inputParams,
 		&m_outputParams,
@@ -646,14 +640,14 @@ void Audio::AudioEngine::restartStream() {
 		NULL );
 
 	if( error != paNoError ) {
-		NanThrowTypeError("Failed to open audio stream :(");
+		Nan::ThrowTypeError("Failed to open audio stream :(");
 	}
 
 	// Start the audio stream
 	error = Pa_StartStream( m_pPaStream );
 
 	if( error != paNoError )
-		NanThrowTypeError("Failed to start audio stream");
+		Nan::ThrowTypeError("Failed to start audio stream");
 
 } // end AudioEngine::restartStream()
 

--- a/NodeCoreAudio/AudioEngine.cpp
+++ b/NodeCoreAudio/AudioEngine.cpp
@@ -10,7 +10,6 @@
 #include "portaudio.h"
 #include <v8.h>
 #include <uv.h>
-#include <node_internals.h>
 #include <node_object_wrap.h>
 #include <stdlib.h>
 

--- a/NodeCoreAudio/AudioEngine.h
+++ b/NodeCoreAudio/AudioEngine.h
@@ -78,14 +78,14 @@ namespace Audio {
 		void queueOutputBuffer( Handle<Array> result );			//!< Queues up an array to be sent to the sound card
 		void setSample( int position, Handle<Value> sample );	//!< Sets a sample in the queued output buffer
 
-		Handle<Array> getInputBuffer();							//!< Returns a v8 array filled with input samples
+		Local<Array> getInputBuffer();							//!< Returns a v8 array filled with input samples
 		Handle<Number> getSample( int position );				//!< Returns a sound card sample converted to a v8 Number
 
 		PaStream *m_pPaStream;				//!< The PortAudio stream object
 		PaStreamParameters m_inputParams,	//!< PortAudio stream parameters
 						   m_outputParams;
 
-		Handle<Array> m_hInputBuffer;		//!< Our pre-allocated input buffer
+		Local<Array> m_hInputBuffer;		//!< Our pre-allocated input buffer
 
 		uv_thread_t ptStreamThread,			//!< Our stream thread
 					jsAudioThread;			//!< Our JavaScript Audio thread

--- a/NodeCoreAudio/AudioEngine.h
+++ b/NodeCoreAudio/AudioEngine.h
@@ -8,7 +8,6 @@
 #include "portaudio.h"
 #include <v8.h>
 #include <vector>
-#include <node_internals.h>
 #include <node_object_wrap.h>
 #include <nan.h>
 using namespace v8; using namespace std;

--- a/NodeCoreAudio/NodeCoreAudio.cpp
+++ b/NodeCoreAudio/NodeCoreAudio.cpp
@@ -27,18 +27,18 @@ NAN_METHOD(CreateEngine)
 void InitAll(Handle<Object> target) {
     Audio::AudioEngine::Init( target );
 
-    NODE_SET_METHOD(target, "createAudioEngine", Audio::AudioEngine::NewInstance);
+    Nan::SetMethod(target, "createAudioEngine", Audio::AudioEngine::NewInstance);
     //NODE_SET_METHOD(target, "createAudioEngine", CreateEngine);
     //target->Set( NanNew<String>("createAudioEngine"), CreateEngine);
     //target->Set( NanNew<String>("createAudioEngine"), NanNew<FunctionTemplate>(CreateEngine)->GetFunction() );
     //target->Set( NanNew<String>("createAudioEngine"), NanNew<FunctionTemplate>(Audio::AudioEngine::NewInstance)->GetFunction() );
     
-    target->Set( NanNew<String>("sampleFormatFloat32"), NanNew<Number>(1) );
-    target->Set( NanNew<String>("sampleFormatInt32"), NanNew<Number>(2) );
-    target->Set( NanNew<String>("sampleFormatInt24"), NanNew<Number>(4) );
-    target->Set( NanNew<String>("sampleFormatInt16"), NanNew<Number>(8) );
-    target->Set( NanNew<String>("sampleFormatInt8"), NanNew<Number>(10) );
-    target->Set( NanNew<String>("sampleFormatUInt8"), NanNew<Number>(20) );
+    target->Set( Nan::New<String>("sampleFormatFloat32").ToLocalChecked(), Nan::New<Number>(1));
+    target->Set( Nan::New<String>("sampleFormatInt32").ToLocalChecked(), Nan::New<Number>(2) );
+    target->Set( Nan::New<String>("sampleFormatInt24").ToLocalChecked(), Nan::New<Number>(4) );
+    target->Set( Nan::New<String>("sampleFormatInt16").ToLocalChecked(), Nan::New<Number>(8) );
+    target->Set( Nan::New<String>("sampleFormatInt8").ToLocalChecked(), Nan::New<Number>(10) );
+    target->Set( Nan::New<String>("sampleFormatUInt8").ToLocalChecked(), Nan::New<Number>(20) );
 }
 
 NODE_MODULE( NodeCoreAudio, InitAll );

--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
   "dependencies": {
     "audio-streamer": ">= 0.1.0",
     "fft": "~0.2.1",
-    "nan": "^1.8.4",
+    "nan": "^2.1.0",
     "node-waveheader": "git://github.com/mrose17/node-waveheader.git",
     "portfinder": "0.2.1"
   },

--- a/package.json
+++ b/package.json
@@ -16,8 +16,11 @@
   },
   "description": "Core native node.js audio functionality, including sound card access and audio streaming",
   "main": "./node-core-audio",
+  "scripts": {
+    "test": "node test/test.js"
+  },
   "engines": {
-    "node": ">=0.8.0"
+    "node": ">=0.12.0"
   },
   "keywords": [
     "audio",

